### DISCUSSION TITLE
Ant runinstall Error

### DISCUSSION THREAD

**Discussion by FaysalMossi** [Keyfactor/ejbca-ce#480] [Q&A] [answered]
pls how to fix this error :
ejbca:install:

ejbca:initCA:
     [echo] Initializing CA with 'ManagementCA' 'CN=ManagementCA,O=EJBCA Sample,C=SE' 'soft' '<ca.tokenpassword hidden>' '2048' 'RSA' '3650' 'null' 'SHA256WithRSA'  -certprofile "ROOTCA" -superadmincn 'SuperAdmin'...
     [java] 14:53:43.131 [main] ERROR org.ejbca.ui.cli.infrastructure.library.CommandLibrary - Error: CLI could not contact EJBCA instance. Either your application server is not up and running, EJBCA has not been deployed successfully, or some firewall rule is blocking the CLI from the application server.
     [java]
     [java] Please be aware that most commands will not work without the application server available.
     [java]
     [java] Exception in thread "main" javax.ejb.NoSuchEJBException: EJBCLIENT000079: Unable to discover destination for request for EJB StatelessEJBLocator for "ejbca/cesecore-ejb/GlobalConfigurationSessionBean", view is interface org.cesecore.configuration.GlobalConfigurationSessionRemote, affinity is None
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:553)
     [java]     at org.jboss.ejb.protocol.remote.RemotingEJBClientInterceptor.handleInvocationResult(RemotingEJBClientInterceptor.java:57)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:624)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:553)
     [java]     at org.jboss.ejb.client.TransactionPostDiscoveryInterceptor.handleInvocationResult(TransactionPostDiscoveryInterceptor.java:148)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:624)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:553)
     [java]     at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.handleInvocationResult(DiscoveryEJBClientInterceptor.java:129)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:624)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:553)
     [java]     at org.jboss.ejb.client.NamingEJBClientInterceptor.handleInvocationResult(NamingEJBClientInterceptor.java:87)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:624)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:553)
     [java]     at org.jboss.ejb.client.TransactionInterceptor.handleInvocationResult(TransactionInterceptor.java:212)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:624)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:553)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.awaitResponse(EJBClientInvocationContext.java:1010)
     [java]     at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:182)
     [java]     at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:116)
     [java]     at com.sun.proxy.$Proxy10.getCachedConfiguration(Unknown Source)
     [java]     at org.ejbca.ui.cli.infrastructure.command.PasswordUsingCommandBase.execute(PasswordUsingCommandBase.java:162)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary$Branch.execute(CommandLibrary.java:309)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary$Branch.execute(CommandLibrary.java:320)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary.findAndExecuteCommandFromParameters(CommandLibrary.java:88)
     [java]     at org.ejbca.ui.cli.EjbcaEjbCli.main(EjbcaEjbCli.java:38)
     [java]     Suppressed: java.net.ConnectException: Connection refused: no further information
     [java]             at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
     [java]             at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
     [java]             at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:327)
     [java]             at org.xnio.nio.WorkerThread.run(WorkerThread.java:591)
     [java]             at ...asynchronous invocation...(Unknown Source)
     [java]             at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:599)
     [java]             at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:565)
     [java]             at org.jboss.remoting3.ConnectionInfo$None.getConnection(ConnectionInfo.java:82)
     [java]             at org.jboss.remoting3.ConnectionInfo.getConnection(ConnectionInfo.java:55)
     [java]             at org.jboss.remoting3.EndpointImpl.doGetConnection(EndpointImpl.java:499)
     [java]             at org.jboss.remoting3.EndpointImpl.getConnectedIdentity(EndpointImpl.java:445)
     [java]             at org.jboss.remoting3.UncloseableEndpoint.getConnectedIdentity(UncloseableEndpoint.java:52)
     [java]             at org.jboss.ejb.protocol.remote.RemotingEJBDiscoveryProvider.getConnectedIdentityUsingClusterEffective(RemotingEJBDiscoveryProvider.java:341)
     [java]             at org.jboss.ejb.protocol.remote.RemotingEJBDiscoveryProvider$DiscoveryAttempt.connectAndDiscover(RemotingEJBDiscoveryProvider.java:435)
     [java]             at org.jboss.ejb.protocol.remote.RemotingEJBDiscoveryProvider.discover(RemotingEJBDiscoveryProvider.java:232)
     [java]             at org.jboss.ejb.protocol.remote.RemoteEJBDiscoveryConfigurator.lambda$configure$0(RemoteEJBDiscoveryConfigurator.java:42)
     [java]             at org.wildfly.discovery.impl.AggregateDiscoveryProvider.discover(AggregateDiscoveryProvider.java:58)
     [java]             at org.wildfly.discovery.ConfiguredProvider.discover(ConfiguredProvider.java:47)
     [java]             at org.wildfly.discovery.Discovery.discover(Discovery.java:122)
     [java]             at org.wildfly.discovery.Discovery.discover(Discovery.java:93)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.discover(DiscoveryEJBClientInterceptor.java:339)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.doAnyDiscovery(DiscoveryEJBClientInterceptor.java:495)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.executeDiscovery(DiscoveryEJBClientInterceptor.java:419)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.handleInvocation(DiscoveryEJBClientInterceptor.java:109)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:516)
     [java]             at org.jboss.ejb.client.NamingEJBClientInterceptor.handleInvocation(NamingEJBClientInterceptor.java:69)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:516)
     [java]             at org.jboss.ejb.client.TransactionInterceptor.handleInvocation(TransactionInterceptor.java:205)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:516)
     [java]             at org.wildfly.common.context.Contextual.runExConsumer(Contextual.java:203)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequestInitial(EJBClientInvocationContext.java:343)
     [java]             at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:178)
     [java]             ... 7 more

BUILD FAILED
C:\ejbca\ejbca-ce\build.xml:71: The following error occurred while executing this line:
C:\ejbca\ejbca-ce\bin\cli.xml:98: The following error occurred while executing this line:
C:\ejbca\ejbca-ce\bin\cli.xml:126: The following error occurred while executing this line:
C:\ejbca\ejbca-ce\bin\cli.xml:211: Java returned: 1

Total time: 6 seconds

**Comment by primetomas** [comment]
The message is this:

"Error: CLI could not contact EJBCA instance. Either your application server is not up and running, EJBCA has not been deployed successfully, or some firewall rule is blocking the CLI from the application server."

Look in your JBoss server log to see that it is actually running.

**Reply by FaysalMossi** [reply]
oh thank you sire

**Reply by FaysalMossi** [reply]
this is what i have in the log :

2024-01-23 14:56:12,137 INFO  [org.ejbca.core.ejb.services.ServiceSessionBean] (EJB default - 1) Timer was triggered. Trying to run service with ID 0.
2024-01-23 14:56:13,023 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 14:56:13,023 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:01:13,047 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 15:01:13,047 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:06:13,055 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 15:06:13,060 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:11:13,060 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 2) Reloading CA certificate cache.
2024-01-23 15:11:13,072 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 2) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:16:13,073 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 15:16:13,073 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:21:13,080 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 15:21:13,080 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:26:13,081 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 15:26:13,081 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates
2024-01-23 15:31:13,091 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloading CA certificate cache.
2024-01-23 15:31:13,091 INFO  [org.cesecore.certificates.certificate.CertificateStoreSessionBean] (EJB default - 1) Reloaded CA certificate cache with 0 certificates

**Reply by primetomas** [reply]
That is a very short snippet of a log. 
But guessing, it looks like something is running.

What version of everything are you running btw? And what commands are you running? I.e. where in the installation guide are you?

**Comment by FaysalMossi** [comment]
I use 
EJBCA 8.2
JBoss 7.4
Ant 1.9
MariaDB 8
and the doc is https://doc.primekey.com/ejbca/ejbca-installation/installation-prerequisites
on Windonws

**Reply by primetomas** [reply, marked as answer]
Then my best guess is that something with the configuration of Remoting failed:
https://doc.primekey.com/ejbca/ejbca-installation/application-servers/jboss-eap-7-4#JBossEAP7.4-ConfigureWildFlyRemoting
